PapyrusCore/Sources/Interceptors/CurlInterceptor.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+/// An `Interceptor` that logs requests based on a condition
+public struct CurlLogger {
+    public enum Condition {
+        case always
+
+        /// only log when the request encountered an error
+        case onError
+    }
+
+    let loggingFn: (String) -> Void
```

Comment:
nit: would you mind making this `log` or `logHandler`? Slightly more in line with other closure names in codebase.

Reply:
fixed